### PR TITLE
SDN cleanup; kill "Registry" type

### PR DESCRIPTION
**Pull request by danwinship**
As discussed elsewhere recently, pkg/sdn/plugin/registry.go is a relic of the old standalone openshift-sdn (and not even in a way that would be useful for a future standalone openshift-sdn), so this gets rid of it (and manages to have less total code in the end).

@openshift/networking PTAL

### CODE REVIEW

File: pkg/sdn/plugin/node.go [hunk 1]
```diff
@@ -79,13 +83,16 @@ func NewNodePlugin(pluginName string, osClient *osclient.Client, kClient *kclien
...
-	var err error
-	node.networkInfo, err = getNetworkInfo(node.registry.oClient)
+	cn, err := node.osClient.ClusterNetwork().Get(osapi.ClusterNetworkDefault)
 	if err != nil {
 		return fmt.Errorf("Failed to get network information: %v", err)
 	}
+	node.networkInfo, err = parseNetworkInfo(cn.Network, cn.ServiceNetwork)
+	if err != nil {
+		return err
+	}
```
Comment (dcbw) [inline pkg/sdn/plugin/node.go]: This all could be a helper function that returns cn, netinfo, and err.  It's also used in proxy.go.

**Comment by dcbw** [comment]
Generally LGTM

File: pkg/sdn/plugin/node.go [hunk 2]
```diff
@@ -124,7 +133,24 @@ func (node *OsdnNode) Start() error {
 }
 
 func (node *OsdnNode) GetLocalPods(namespace string) ([]kapi.Pod, error) {
-	return node.registry.GetRunningPods(node.hostName, namespace)
+	fieldSelector := fields.Set{"spec.host": node.hostName}.AsSelector()
```
Comment (pravisankar) [inline pkg/sdn/plugin/node.go]: "spec.hostname"

File: pkg/sdn/plugin/plugin.go [hunk 3]
```diff
@@ -153,13 +155,32 @@ func getScriptError(output []byte) string {
 	return string(output)
 }
 
+func (plugin *OsdnNode) getPod(nodeName, namespace, podName string) (*kapi.Pod, error) {
+	fieldSelector := fields.Set{"spec.host": nodeName}.AsSelector()
```
Comment (pravisankar) [inline pkg/sdn/plugin/plugin.go]: "spec.hostname"

File: pkg/sdn/plugin/proxy.go [hunk 4]
```diff
@@ -41,25 +43,34 @@ func NewProxyPlugin(pluginName string, osClient *osclient.Client, kClient *kclie
...
 	glog.Infof("Starting multitenant SDN proxy endpoint filter")
 
+	cn, err := proxy.osClient.ClusterNetwork().Get(osapi.ClusterNetworkDefault)
+	if err != nil {
+		return fmt.Errorf("could not get network info: %s", err)
+	}
+	proxy.networkInfo, err = parseNetworkInfo(cn.Network, cn.ServiceNetwork)
+	if err != nil {
+		return err
+	}
```
Comment (pravisankar) [inline pkg/sdn/plugin/proxy.go]: As @dcbw suggested, we could add a helper method to fetch and parse networkInfo.

**Comment by pravisankar** [comment]
Overall LGTM

File: pkg/sdn/plugin/node.go (hunk 2 above)
Comment (danwinship) [inline pkg/sdn/plugin/node.go]: hm... actually, it's referring to a deprecated v1.PodSpec field which was replaced with "nodeName", not "hostname", which is apparently something slightly different.

Comment (liggitt) [inline pkg/sdn/plugin/node.go]: that's the same field (deprecated version and current version), though the assumption that it is a hostname has always been tenuous.

**Comment by danwinship** [comment]
added back the removed getNetworkInfo() function and added a new commit fixing the deprecated field name.
[test]

**Comment by smarterclayton** [comment]
With the networking tests disabled, do you have a clean local run of the extended suite?

**Comment by danwinship** [comment]
> With the networking tests disabled, do you have a clean local run of the extended suite?

test/extended/conformance.sh doesn't pass locally even in master

**Comment by danwinship** [comment]
[test][extended]

**Comment by danwinship** [comment]
https://ci.openshift.redhat.com/jenkins/job/test_pr_origin/8995/ flakes #9490 and #10951
[test]

**Comment by marun** [comment]
It should be possible to get a clean local run with #10972.  A clean ci run won't be possible until I figure out the cert issue.

**Comment by danwinship** [comment]
after (locally) rebasing on top of #11039, extended networking tests pass

**Comment by dcbw** [comment]
Changes for my comments LGTM and this is tested.  [merge]

**Comment by dcbw** [comment]
flakes all around:
https://github.com/openshift/origin/issues/9548
https://github.com/openshift/origin/issues/9490
https://github.com/openshift/origin/issues/10765
https://github.com/openshift/origin/issues/11008

**Comment by dcbw** [comment]
[merge]

**Comment by dcbw** [comment]
flake city [merge]

**Comment by dcbw** [comment]
Deploymentconfig flake https://github.com/openshift/origin/issues/11016 again

**Comment by danwinship** [comment]
you have to say [merge] again when you identify the flake #11016 or openshift-bot won't realize you're talking to it

**Comment by dcbw** [comment]
@danwinship yeah I know, but I was going to hold off until something is done about #11016 since it's killing a lot of our PRs right now.  Until that gets fixed I don't think another merge will be useful.

**Comment by danwinship** [comment]
*some* merges have made it through... but anyway, it looks like we didn't need a re-tag anyway because it's still in the merge queue

**Pull request merged**